tests/readmes/test_erc20_enterprise.py

```diff
@@ -0,0 +1,266 @@
+        config.transaction_timeout,
+    )
+    assert alice_wallet.address
+
+    # Mint OCEAN
+    mint_fake_OCEAN(config)
+    assert alice_wallet.web3.eth.get_balance(alice_wallet.address) > 0, "need ETH"
+
+    # Publish an NFT token
+    nft_token = ocean.create_nft_token("NFTToken1", "NFT1", alice_wallet)
+    token_address = nft_token.address
+    assert token_address
+
+    # Prepare data for ERC20 Enterprise token
+    cap = ocean.to_wei(200)
+    erc20_data = CreateErc20Data(
+        template_index=2,  # default value
+        strings=["ERC20DT1", "ERC20DT1Symbol"],  # name & symbol for ERC20 token
+        addresses=[
+            alice_wallet.address,  # minter address
+            alice_wallet.address,  # fee manager for this ERC20 token
+            alice_wallet.address,  # publishing Market Address
+            ZERO_ADDRESS,  # publishing Market Fee Token
+        ],
+        uints=[cap, 0],
+        bytess=[b""],
+    )
+    erc20_enterprise_token = nft_token.create_datatoken(
+        erc20_data=erc20_data, from_wallet=alice_wallet
+    )
+
+    assert erc20_enterprise_token.address
+    assert erc20_enterprise_token.token_name() == "ERC20DT1"
+    assert erc20_enterprise_token.symbol() == "ERC20DT1Symbol"
+
+    bob_private_key = os.getenv("TEST_PRIVATE_KEY2")
+    bob_wallet = Wallet(
+        ocean.web3,
+        bob_private_key,
+        config.block_confirmations,
+        config.transaction_timeout,
+    )
+
+    # Verify that Bob has ganache ETH
+    assert ocean.web3.eth.get_balance(bob_wallet.address) > 0, "need ganache ETH"
+
+    # Bob gets 1 DT from dispenser and then startsOrder, while burning that DT
+    dispenser = ocean.dispenser
+    dispenser_data = DispenserData(
+        dispenser_address=dispenser.address,
+        allowed_swapper=ZERO_ADDRESS,
+        max_balance=ocean.to_wei(50),
+        with_mint=True,
+        max_tokens=ocean.to_wei(50),
+    )
+    tx = erc20_enterprise_token.create_dispenser(
+        dispenser_data=dispenser_data, from_wallet=alice_wallet
+    )
+    assert tx, "Dispenser not created!"
+
+    OCEAN_token = ocean.get_datatoken(ocean.OCEAN_address)
+    consume_fee_amount = ocean.to_wei(2)
+    erc20_enterprise_token.set_publishing_market_fee(
+        publish_market_fee_address=bob_wallet.address,
+        publish_market_fee_token=OCEAN_token.address,  # can be also USDC, DAI
+        publish_market_fee_amount=consume_fee_amount,
+        from_wallet=alice_wallet,
+    )
+
+    # Approve tokens.
+    OCEAN_token.approve(
+        spender=erc20_enterprise_token.address,
+        amount=consume_fee_amount,
+        from_wallet=alice_wallet,
+    )
+
+    # Prepare data for order.
+    provider_fee_address = alice_wallet.address
+    provider_fee_token = OCEAN_token.address
+    provider_fee_amount = 0
+    provider_data = json.dumps({"timeout": 0}, separators=(",", ":"))
+    valid_until = 1958133628  # 2032
+    signature = ocean.compute_signature(
```

Comment:
is it possible to build the entire ProviderFee structure instead of just the signature? Then Wee can pass that as an argument to OrderParams directly, instead of re-packaging it as a tuple.